miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java

```diff
@@ -588,6 +594,20 @@ private void validateChange(Library library, Library beforeChange) throws IOExce
     }
   }
 
+  private void validateTargetedSequencing(LibraryDesignCode libraryDesignCode, Collection<LibraryDilution> libraryDilutions,
+      List<ValidationError> errors) throws IOException {
+    libraryDesignCode = libraryDesignCodeService.get(libraryDesignCode.getId());
+    if (libraryDesignCode.isTargetedSequencingRequired()) {
+      List<String> badDilutions = libraryDilutions.stream().filter(dilution -> dilution.getTargetedSequencing() == null)
+          .map(LibraryDilution::getName).collect(Collectors.toList());
+      if (!badDilutions.isEmpty()) {
+        errors.add(new ValidationError("libraryDesignCodeId",
```

Comment:
Property should be "libraryDesignCode" rather than "libraryDesignCodeId"

Reply:
I'm unclear on these. This isn't the name of the column nor the name of the DTO field. Is this the name in the JSP?